server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -506,6 +506,11 @@ public synchronized IndexShard createShard(
                         }
                         logger.info("DocRep shard {} is migrating to remote", shardId);
                         seedRemote = true;
+                    } else if (sourceNode == null && routing.primary()) {
+                        // seeding when index metadata says so
```

Comment:
we can remove `routing.primary()` check as if sourceNode is null then it has to be primary . We can throw `IllegalStateException` if it is not just like above .

Reply:
Can we check if for a `replica shard`'s create Shard , `sourceNode`always has to be there  ? Even if a node having replica starts  . We can write an IT to verify the same.